### PR TITLE
General code cleanup

### PR DESCRIPTION
**Pull request by pranavkm**
* Removing NotNullAttribute
* Remove warnings and unused code
* Replacing Contract.Assert with Debug.Assert

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @Eilon \ @NTaylorMullen

**Comment by pranavkm** [comment]
cc @barrytang

File: src/Microsoft.Extensions.CodeGeneration.Core/CodeGeneratorActionsService.cs [hunk 1]
```diff
@@ -55,9 +55,14 @@ public class CodeGeneratorActionsService : ICodeGeneratorActionsService
         }
 
         public async Task AddFileFromTemplateAsync(string outputPath, string templateName,
-            [NotNull]IEnumerable<string> templateFolders,
-            [NotNull]object templateModel)
+            IEnumerable<string> templateFolders,
+            object templateModel)
```
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.CodeGeneration.Core/CodeGeneratorActionsService.cs]: `templateModel` is ok to be `null`?

File: src/Microsoft.Extensions.CodeGeneration.Sources/NotNullArgument.cs [hunk 2]
```diff
@@ -1,12 +0,0 @@
-// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-using System;
-
-namespace Microsoft.Extensions.CodeGeneration
-{
-    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
-    internal sealed class NotNullAttribute : Attribute
```
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.CodeGeneration.Sources/NotNullArgument.cs]: :v:

File: src/Microsoft.Extensions.CodeGenerators.Mvc/Controller/MvcControllerEmpty.cs [hunk 3]
```diff
@@ -13,13 +13,33 @@ namespace Microsoft.Extensions.CodeGenerators.Mvc.Controller
...
+            ICodeGeneratorActionsService codeGeneratorActionsService,
+            IServiceProvider serviceProvider,
+            ILogger logger)
             : base(libraryManager, environment, codeGeneratorActionsService, serviceProvider, logger)
         {
+            if (libraryManager == null)
+            {
+                throw new ArgumentNullException(nameof(libraryManager));
+            }
+            if (environment == null)
```
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.CodeGenerators.Mvc/Controller/MvcControllerEmpty.cs]: just to be consistent with everything else you've done here separate each of em by empty lines.

File: src/Microsoft.Extensions.CodeGenerators.Mvc/Dependency/MvcLayoutDependencyInstaller.cs [hunk 4]
```diff
@@ -15,13 +15,33 @@ namespace Microsoft.Extensions.CodeGenerators.Mvc.Dependency
...
+            ILogger logger,
+            IPackageInstaller packageInstaller,
+            IServiceProvider serviceProvider)
             : base(libraryManager, applicationEnvironment, logger, packageInstaller, serviceProvider)
         {
+            if (libraryManager == null)
+            {
+                throw new ArgumentNullException(nameof(libraryManager));
+            }
+            if (applicationEnvironment == null)
```
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.CodeGenerators.Mvc/Dependency/MvcLayoutDependencyInstaller.cs]: ditto

File: src/Microsoft.Extensions.CodeGenerators.Mvc/Dependency/StaticFilesDependencyInstaller.cs [hunk 5]
```diff
@@ -16,13 +16,33 @@ namespace Microsoft.Extensions.CodeGenerators.Mvc.Dependency
...
+            ILogger logger,
+            IPackageInstaller packageInstaller,
+            IServiceProvider serviceProvider)
             : base(libraryManager, applicationEnvironment, logger, packageInstaller, serviceProvider)
         {
+            if (libraryManager == null)
+            {
+                throw new ArgumentNullException(nameof(libraryManager));
+            }
+            if (applicationEnvironment == null)
```
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.CodeGenerators.Mvc/Dependency/StaticFilesDependencyInstaller.cs]: +1

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by Eilon** [comment]
:shipit:

File: src/Microsoft.Extensions.CodeGeneration.Core/CodeGeneratorActionsService.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.Extensions.CodeGeneration.Core/CodeGeneratorActionsService.cs]: A test invoked it with `null` and expected it to pass.

**Comment by barrytang** [comment]
Adding @abpiskunov and @prafullbhosale.

**Pull request closed without merging**